koku/api/report/aws/openshift/query_handler.py

```diff
@@ -75,6 +75,10 @@ def execute_query(self):  # noqa: C901
             order_date = None
             for i, param in enumerate(query_order_by):
                 if check_if_valid_date_str(param):
+                    # Checks to see if the date is in the query_data
+                    if not any(d["date"] == param for d in query_data):
+                        break
+                    # else we set order_date to a valid date
                     order_date = param
                     break
```

Comment:
**Question**

I'm wondering if this would be more efficient:

```suggestion
            order_date = None
            for i, param in enumerate(query_order_by):
                if check_if_valid_date_str(param):
                    for d in query_data:
                        found = d["date"] == param
                        if found:
                            order_date = param
                            break
                    if order_date is not None:
                        break
```

The intent here would be to stop on the first hit instead of running though the entire query_data list every time we hit a date.

Also, it seems to me that this is replicated a lot. Looks like a good candidate to be put into a common module.

Reply:
I changed the function to 
                if check_if_valid_date_str(param):
                    # Checks to see if the date is in the query_data
                    if any(d["date"] == param for d in query_data):
                        # Set order_date to a valid date
                        order_date = param
                        break
I think this is the most efficient method

Reply:
So long as `any` will short-circuit on the first true-y value, then yes.

Reply:
https://docs.python.org/3/library/functions.html#any

Reply:
Given with Andrews comment the any() will achieve the same goal